### PR TITLE
executors/bf: suppress instcombine error for llvm >=18

### PR DESCRIPTION
**Pull request by int-y1**
bf is broken due to
```
dmoj.error.CompileError: LLVM ERROR: Instruction Combining did not reach a fixpoint after 1 iterations
```

use the `instcombine<no-verify-fixpoint>` trick from https://github.com/llvm/llvm-project/issues/92648 to suppress this error

```
0.929 Testing BF:     Success                                                                                                                                                                              
1.201   opt: 19.1.6                                                                                                                                                                                        
1.214   llc: 19.1.6                                                                                                                                                                                        
1.214   ld_x64: 2.43.50.20241215
1.214 
1.214 Configuration result:
1.214 runtime:
1.214   ld_x64: /usr/bin/x86_64-linux-gnu-ld
1.214   llc: /usr/bin/llc
1.214   opt: /usr/bin/opt
```

~~unfortunately, `'instcombine<no-verify-fixpoint>'` doesn't exist in llvm 16, and this is causing ci to fail~~

for llvm 17 and earlier, use `instcombine`. for llvm 18 and later, use `instcombine<no-verify-fixpoint>`.

### CODE REVIEW

File: dmoj/executors/BF.py [hunk 1]
```diff
@@ -185,15 +185,15 @@ def compile_to_llvm(source_code: bytes) -> bytes:
 
 OPT_PASSES = [
     # Shorten instructions
-    'instcombine',
+    'instcombine<no-verify-fixpoint>',  # https://github.com/llvm/llvm-project/issues/92648
```
Comment (quantum5) [inline dmoj/executors/BF.py:188]: We know the llvm version because we use it for reporting. Can we just detect it here?

Comment (int-y1) [reply dmoj/executors/BF.py]: done

opt 17.0.1 works with `instcombine` and opt 18.1.0 works with `instcombine<no-verify-fixpoint>`, so i set the cutoff to 18

**Review by quantum5 — APPROVED** [review]

**Comment by quantum5** [comment]
ok to test

**Comment by quantum5** [comment]
rerun freebsd ci

**Comment by quantum5** [comment]
rerun freebsd ci

**Pull request merged**